File: SDKs/twilio.com/twilio_events_v1/1.36.0/go/pkg/models/operations/deletesubscribedevent.go
package operations

import (
"openapi/pkg/models/shared")
var DeleteSubscribedEventServers = []string{
	"https://events.twilio.com",
}

type DeleteSubscribedEventPathParams struct {
    SubscriptionSid string `pathParam:"style=simple,explode=false,name=SubscriptionSid"`
    Type string `pathParam:"style=simple,explode=false,name=Type"`
    
}

type DeleteSubscribedEventSecurity struct {
    AccountSidAuthToken shared.SchemeAccountSidAuthToken `security:"scheme,type=http,subtype=basic"`
    
}

type DeleteSubscribedEventRequest struct {
    ServerURL *string 
    PathParams DeleteSubscribedEventPathParams 
    Security DeleteSubscribedEventSecurity 
    
}

type DeleteSubscribedEventResponse struct {
    ContentType string 
    StatusCode int64 
    
}

File: SDKs/twilio.com/twilio_events_v1/1.37.3/typescript/src/sdk/models/operations/deletesubscribedevent.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";

export const DeleteSubscribedEventServerList = [
	"https://events.twilio.com",
] as const;


export class DeleteSubscribedEventPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=SubscriptionSid" })
  subscriptionSid: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=Type" })
  type: string;
}


export class DeleteSubscribedEventSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=http;subtype=basic" })
  accountSidAuthToken: shared.SchemeAccountSidAuthToken;
}


export class DeleteSubscribedEventRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  serverUrl?: string;

  @SpeakeasyMetadata()
  pathParams: DeleteSubscribedEventPathParams;

  @SpeakeasyMetadata()
  security: DeleteSubscribedEventSecurity;
}


export class DeleteSubscribedEventResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}
